src/benchmark.cc

```diff
@@ -684,505 +472,480 @@ void Benchmark::AddRange(std::vector<int>* dst, int lo, int hi, int mult) {
   // Add "lo"
   dst->push_back(lo);
 
+  static const int kint32max = std::numeric_limits<int32_t>::max();
+
   // Now space out the benchmarks in multiples of "mult"
-  for (int32_t i = 1; i < std::numeric_limits<int32_t>::max() / mult;
-       i *= mult) {
+  for (int32_t i = 1; i < kint32max/mult; i *= mult) {
     if (i >= hi) break;
-    if (i > lo) dst->push_back(i);
+    if (i > lo) {
+      dst->push_back(i);
+    }
   }
   // Add "hi" (if different from "lo")
-  if (hi != lo) dst->push_back(hi);
+  if (hi != lo) {
+    dst->push_back(hi);
+  }
 }
 
-std::vector<Benchmark::Instance> Benchmark::CreateBenchmarkInstances(
-    size_t rangeXindex, size_t rangeYindex) {
-  // Special list of thread counts to use when none are specified
-  std::vector<int> one_thread;
-  one_thread.push_back(1);
+} // end namespace internal
 
-  std::vector<Benchmark::Instance> instances;
-
-  const bool is_multithreaded = (!thread_counts_.empty());
-  const std::vector<int>& thread_counts =
-      (is_multithreaded ? thread_counts_ : one_thread);
-  for (int num_threads : thread_counts) {
-    Instance instance;
-    instance.name = name_;
-    instance.bm = this;
-    instance.threads = num_threads;
-
-    if (rangeXindex != kNoRangeIndex) {
-      instance.rangeX = rangeX_[rangeXindex];
-      instance.rangeXset = true;
-      AppendHumanReadable(instance.rangeX, &instance.name);
-    }
-    if (rangeYindex != kNoRangeIndex) {
-      instance.rangeY = rangeY_[rangeYindex];
-      instance.rangeYset = true;
-      AppendHumanReadable(instance.rangeY, &instance.name);
-    }
+namespace {
 
-    // Add the number of threads used to the name
-    if (is_multithreaded) {
-      std::stringstream ss;
-      ss << "/threads:" << instance.threads;
-      instance.name += ss.str();
-    }
+bool running_benchmark = false;
 
-    instances.push_back(instance);
-  }
+// Global variable so that a benchmark can cause a little extra printing
+std::string report_label GUARDED_BY(GetBenchmarkLock());
 
-  return instances;
+// Should this benchmark base decisions off of real time rather than
+// cpu time?
+bool use_real_time GUARDED_BY(GetBenchmarkLock());
+
+// Return prefix to print in front of each reported line
+static const char* Prefix() {
```

Comment:
aside: this should probably be printed in the header info, not per run. maybe an issue is needed - what do you think?